### PR TITLE
9.get /api/reviews/review id/comments

### PR DESCRIPTION
**Pull request by DeanH1990**
This PR adds getCommentsByReviewId function to app, controller and model. 

The happy path tests that the returned object is an array, in newest order first. 

The sad path tests check if the review id is invalid, valid review id type but does not exist and valid review id type that does exist but has no comments.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -164,6 +164,65 @@ describe('app', () => {
...
+                        })
+                        test('status 200: responds with newest comments first', () => {
+                            return request(app)
+                            .get('/api/reviews/2/comments')
+                            .expect(200)
+                            .then(({ body }) => {
+                                const { comments } = body;
+                                expect(comments).toBeSortedBy('created_at', { descending: true });
+                            })
+                        })
```
Comment (corkesw) [inline __tests__/app.test.js:266]: You could add a test to check what happens when we request an article with no comments - I'd expect a 200 status and an empty array to be returned

File: models/app.models.js [hunk 2]
```diff
@@ -43,4 +43,18 @@ exports.patchReviewVotesById = (review_id, inc_votes) => {
...
\ No newline at end of file
+}
+
+exports.selectCommentsByReviewId = (review_id) => {
+    return db.query(`
+    SELECT comments.* 
+    FROM comments
+    WHERE review_id = $1
+    ORDER BY created_at DESC`, [review_id]).then(({ rows: comments }) => {
+        if (comments.length === 0) {
```
Comment (corkesw) [inline models/app.models.js]: Does this logic distinguish between an article with no comments and an article that does not exist?

**Review by corkesw — COMMENTED** [review]
Great work - one more test is needed here and you will have to look at the logic in your model in order to deal with this.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -231,6 +231,67 @@ describe('app', () => {
                         })
                     })
                 })
+                describe('/comments', () => {
+                    describe.only('GET: /api/reviews/:review_id/comments', () => {
```
Comment (jimstev2447) [inline __tests__/app.test.js:235]: watch out for this .only
